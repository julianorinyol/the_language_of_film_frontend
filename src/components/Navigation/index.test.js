import React from 'react';
import { Navigation } from './'
import { shallow, mount } from '../../test/config/enzyme'
import { MemoryRouter } from 'react-router-dom'

const props = {
    location: { 
        pathname: '/films/'
    } 
}

describe('<Navigation />', () => {
    it('should mount', (done) => {
        expect.assertions(1)
    
        const wrap = shallow(<Navigation {...props}/>)

        expect(wrap.exists()).toBeTruthy()
        return done()
    })

    it('should call useState', (done) => {
        expect.assertions(2)
        const setState = jest.fn();
        const useStateSpy = jest.spyOn(React, 'useState')
        useStateSpy.mockImplementation((init) => [init, setState]);

        const wrap = mount(<MemoryRouter>
                <Navigation {...props} />
            </MemoryRouter>
        );
        
        const nav = wrap.find(Navigation)
        
        expect(nav).toMatchSnapshot()

        const newPathname = "/study/"
        
        const wordsTab = wrap.find(`a[href="${newPathname}"]`)
        
        wordsTab.simulate('click');
        expect(setState).toHaveBeenCalledWith(newPathname);
        jest.clearAllMocks();
        return done()
    })
})
