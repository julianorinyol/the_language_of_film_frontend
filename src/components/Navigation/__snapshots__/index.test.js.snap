// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should call useState 1`] = `ReactWrapper {}`;
